app/models/account/master_methods.rb

```diff
@@ -24,12 +28,16 @@ def master_on_premises?
   end
 
   module ClassMethods
+    def find_by_service_token!(service_token, error: ActiveRecord::RecordNotFound)
+      by_service_token(service_token).first || raise(error)
```

Comment:
Is it possible to have multiple results with the same service token?

Reply:
yep, sounds better.

Reply:
actually, in some cases `.first` returns all records.. https://github.com/rails/rails/issues/13647

there's a way how to do it, but i would rather keep it this way..

`find_by_service_token` means `find` and `by_service_token` just an another scope.

Reply:
`by_service_token` supposed to scope collection and return "many". `find_by_service_token!` supposed to return one object or raise error. i think it is consistent.